### PR TITLE
refacto core

### PR DESCRIPTION
**Pull request by bousca**

### CODE REVIEW

File: demo/demo.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const Crowdrz = require('../lib/crowdrz');
+
+const apiKey = 'EAAGnFkxIcgQBAMehzed3niJfZAJINBhT86ryKqqkhQ7qAI4asVTTZCKIJwDFfQ9v4nRdlpF7uullM6zCmAt09hnRtlalzs7ENZAWAdyY3QKN8uFKCPHZBPUc4ghQdiM4i47w83FiZCYUeUrQzcKfCPtJcepzaZCa24FqUjZBveOlyl7J5SIoUBS5RoBn6hFaSd2gYTEw1uo0QZDZD';
```
Comment (benomite) [inline demo/demo.js]: Did you just commited an api key? ^^
You should handle that with env vars

File: lib/crowdrz.js [hunk 2]
```diff
@@ -1,37 +1,108 @@
...
-    } else if (!apiToken) {
-      throw new Error(`Crowdrz JS - ${scope} token are required`);
+  constructor (provider='', apiToken='') {
+    this._resetQuery()
+    this.provider = provider
+    this.apiToken = apiToken
+  }
+
+  // Private method
+  _resetQuery() {
```
Comment (benomite) [inline lib/crowdrz.js]: I’m pretty sure you can cleanly declare private methods in ES6 (you should confirm that before changing your code). Check that out: https://www.sitepoint.com/javascript-private-class-fields/

File: lib/crowdrz.js [hunk 3]
```diff
@@ -1,37 +1,108 @@
...
+    this.setMethod(method)
+      .setEndpoint(endpoint)
+      .setParams(params)
+    const data = await servicesList[this.provider].getData({ apiToken: this.apiToken, ...this.request })
+    this._resetQuery()
+    return data
+    
+  }
+
+  async get(endpoint, params) {
```
Comment (benomite) [inline lib/crowdrz.js]: Use `this.call()` instead of having (almost) duplicated lines

File: lib/crowdrz.js [hunk 4]
```diff
@@ -1,37 +1,108 @@
...
+    this.setMethod('GET')
+      .setEndpoint(endpoint)
+      .setParams(params)
+    const data = await servicesList[this.provider].getData({ apiToken: this.apiToken, ...this.request })
+    this._resetQuery()
+    return data
+  }
+
+  async post(endpoint, params) {
```
Comment (benomite) [inline lib/crowdrz.js]: Use this.call() instead of having (almost) duplicated lines

File: lib/crowdrz.js [hunk 5]
```diff
@@ -1,37 +1,108 @@
...
+    this.setMethod('POST')
+      .setEndpoint(endpoint)
+      .setParams(params)
+    const data = await servicesList[this.provider].getData({ apiToken: this.apiToken, ...this.request })
+    this._resetQuery()
+    return data
+  }
+
+  async put(endpoint, params) {
```
Comment (benomite) [inline lib/crowdrz.js]: Use this.call() instead of having (almost) duplicated lines

File: lib/crowdrz.js [hunk 6]
```diff
@@ -1,37 +1,108 @@
...
+    this.setMethod('PUT')
+      .setEndpoint(endpoint)
+      .setParams(params)
+    const data = await servicesList[this.provider].getData({ apiToken: this.apiToken, ...this.request })
+    this._resetQuery()
+    return data
+  }
+
+  async delete(endpoint, params) {
```
Comment (benomite) [inline lib/crowdrz.js]: Use this.call() instead of having (almost) duplicated lines

File: lib/services/facebook/index.js [hunk 7]
```diff
@@ -2,46 +2,57 @@
...
+    return false
+  }
+  return true
 }
 
-const getDataRow = async (apiRoute, params, dataLimit=25, _res=[]) => {
-  // Delete limit params to avoid bug.
-  if (params.limit) {
-    delete params.limit
+const getData = async (request, _res=[]) => {
```
Comment (benomite) [inline lib/services/facebook/index.js]: I’m not fond of `_res` as a var. It is not explicit and I don’t understand why it start with a `_`

File: lib/services/facebook/index.js [hunk 8]
```diff
@@ -2,46 +2,57 @@
 
 const FB = require('fb').default;
 
-const setToken = apiToken => {
+let totalLimit = null;
+
+const _setToken = apiToken => {
```
Comment (benomite) [inline lib/services/facebook/index.js]: why does this function starts with a `_` ?

Comment (benomite) [reply lib/services/facebook/index.js]: And is it really useful as it is just 1 line?

File: lib/crowdrz.js (hunk 2 above)
Comment (bousca) [reply lib/crowdrz.js]: #resetQuery() {} doesn't work, we need transpiler like babel to use it.

File: lib/services/facebook/index.js (hunk 8 above)
Comment (bousca) [reply lib/services/facebook/index.js]: > why does this function starts with a `_` ?

_  is frequently used to preface the name of an object's property or method that is private.

**Pull request merged**